controllers/index.js

```diff
@@ -113,5 +115,34 @@ function feed(req, res){
 		}).catch((error) => reject(error));
 	});
  }
-module.exports = { login, logout, checkUserAuthentication, viewProfile, addPost, createPost, feed};
 
+ function signup(req, res) {
+	if (req.session.user) {
+		res.redirect("/feed");
+	} else {
+		res.render("signup")
+	}
+}
+
+
+function registerUser(req, res) {
+	if (req.session.user) {
+		res.redirect("/feed");
+	}
+	else{
+		User.create({
+			username: req.body.username,
+			firstname: req.body.firstName,
+			lastname: req.body.lastName,
```

Comment:
I think key in model are in camelCase   like firstName not firstname 

Is it even running ?